### PR TITLE
Fix two left-over usages of cfg(addr_of).

### PR DESCRIPTION
**Pull request by adamreichold**

### CODE REVIEW

**Review by alex — APPROVED** [review]

File: src/types/mod.rs [hunk 1]
```diff
@@ -189,13 +189,6 @@ macro_rules! pyobject_native_type_info(
...
             #[inline]
             fn type_object_raw(_py: $crate::Python<'_>) -> *mut $crate::ffi::PyTypeObject {
-                // Create a very short lived mutable reference and directly
-                // cast it to a pointer: no mutable references can be aliasing
-                // because we hold the GIL.
-                #[cfg(not(addr_of))]
-                unsafe { &mut $typeobject }
-
-                #[cfg(addr_of)]
                 unsafe { ::std::ptr::addr_of_mut!($typeobject) }
```
Comment (davidhewitt) [inline src/types/mod.rs:192]: Do we even need `unsafe` around `addr_of_mut`? I wonder if that was purely there because we needed a block.

Comment (adamreichold) [reply src/types/mod.rs:192]: The `unsafe` is necessary as we still mutably access `static` variables and something also to call `assume_gil_acquired`.

Comment (adamreichold) [reply src/types/mod.rs:192]: (Personally, I find the call to `assume_gil_acquired` untidy, but we would have to rework the macros here to pass the GIL available here into the `$typeobject` expression which sometimes really is just a global and sometimes calls a function to lazily initialize a type object.)

Comment (davidhewitt) [reply src/types/mod.rs:192]: Agreed, refactoring here is welcome though maybe not my first priority. I also have a vague feeling that `#[pyclass]` types will need significant adjusting here if we ever try to remove the statics.

**Review by davidhewitt — APPROVED** [review]
Thanks!

**Pull request merged**